extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -196,7 +196,11 @@ void configureRuntimeProperties(AgroalRecorder recorder,
 
         recorder.configureRuntimeProperties(agroalRuntimeConfig);
 
-        dataSourceInitialized.produce(new DataSourceInitializedBuildItem());
+        Set<String> dataSourceNames = new HashSet<>(agroalBuildTimeConfig.namedDataSources.keySet());
+        if (agroalBuildTimeConfig.defaultDataSource.driver.isPresent()) {
+            dataSourceNames.add("");
```

Comment:
This looks odd, is it necessary?

Reply:
I changed the code to express the intent more clearer.

Flyway needs to know all defined datasource names to support them.
They are collected using a new Collection inside the already existing DataSourceInitializedBuildItem.
To express the presence of the default datasource, an empty String is added to the collection of datasource names.